stl/inc/atomic

```diff
@@ -131,6 +131,74 @@ _NODISCARD extern "C" bool __cdecl __std_atomic_has_cmpxchg16b() noexcept;
 #define ATOMIC_LLONG_LOCK_FREE    2
 #define ATOMIC_POINTER_LOCK_FREE  2
 
+#if _HAS_CXX20
+extern "C" {
+void _CRT_SATELLITE_1 __stdcall __std_atomic_wait_direct(const void* const _Storage, const void* const _Comparand,
+    const std::size_t _Size, unsigned long long& _Wait_context) noexcept;
+void _CRT_SATELLITE_1 __stdcall __std_atomic_notify_one_direct(const void* const _Storage) noexcept;
+void _CRT_SATELLITE_1 __stdcall __std_atomic_notify_all_direct(const void* const _Storage) noexcept;
+
+void _CRT_SATELLITE_1 __stdcall __std_atomic_unwait_direct(
+    const void* const _Storage, unsigned long long& _Wait_context) noexcept;
+
+void _CRT_SATELLITE_1 __stdcall __std_atomic_wait_indirect(
+    const void* const _Storage, unsigned long long& _Wait_context) noexcept;
+void _CRT_SATELLITE_1 __stdcall __std_atomic_notify_one_indirect(const void* const _Storage) noexcept;
+void _CRT_SATELLITE_1 __stdcall __std_atomic_notify_all_indirect(const void* const _Storage) noexcept;
+
+void _CRT_SATELLITE_1 __stdcall __std_atomic_unwait_indirect(
+    const void* const _Storage, unsigned long long& _Wait_context) noexcept;
+
+std::size_t _CRT_SATELLITE_1 __stdcall __std_atomic_get_spin_count(const bool _Is_direct) noexcept;
+
+enum _Atomic_spin_phase : unsigned long long {
+    _Atomic_wait_phase_mask            = 0x0000'000F,
+    _Atomic_spin_value_mask            = 0xFFFF'FFF0,
+    _Atomic_spin_value_step            = _Atomic_wait_phase_mask + 1,
+    _Atomic_counter_value_mask         = ~_Atomic_wait_phase_mask,
+    _Atomic_counter_value_step         = _Atomic_wait_phase_mask + 1,
+    _Atomic_wait_phase_init_spin_count = 0x0000'0000,
+    _Atomic_wait_phase_spin            = 0x0000'0008,
+    _Atomic_wait_phase_wait_locked     = 0x0000'0001,
+    _Atomic_wait_phase_wait_none       = 0x0000'0002,
+    _Atomic_wait_phase_wait_counter    = 0x0000'0004,
+    _Atomic_unwait_needed              = _Atomic_wait_phase_wait_locked,
+};
+
+#if defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0602) || defined(_M_ARM) || defined(_M_ARM64)
+inline constexpr bool _Atomic_wait_on_address_statically_avaliable = true;

```

Comment:
This is never before seen tech for us. Stephan/Casey, please double check strongly.

Also: Should this be `detect_mismatch`'d?

Reply:
This is a configuration setting so it should be locked by `yvals.h`.

Reply:
> Also: Should this be detect_mismatch'd?

Good idea. Honestly, I wasn't aware about the existence of `detect_mismatch`.

> This is never before seen tech for us.

Well, to me as well, but you asked to avoid extra function calls... Do you have something other in mind?

Reply:
I've tried that out. 

`defined(_WIN32_WINNT) && (_WIN32_WINNT >= 0x0602)` looks like a bad idea.
Sometimes `_WIN32_WINNT` is defined to something default for a particular file, even while targeting non-default OS level. Even STL itself compiles with `_STL_WIN32_WINNT` defined to `0x0501` whereas `_WIN32_WINNT` is default.

So I've kept only `defined(_M_ARM) || defined(_M_ARM64)` part